src/app/beer_garden/router.py

```diff
@@ -176,14 +176,28 @@ def route(operation: Operation):
         )
 
     # Determine which garden the operation is targeting
-    if not operation.target_garden_name:
-        operation.target_garden_name = _determine_target_garden(operation)
+    target_garden = _determine_target_garden(operation)
 
-    if not operation.target_garden_name:
+    if not target_garden and not operation.target_garden_name:
```

Comment:
I can't help but feel that there's a lot of work that comes after the `_determine_target_garden` call that looks very much related to determining the target garden.  Is there a reason why most of this logic shouldn't live there?

Reply:
Yeah, it should probably be renamed / organized better at this point.